src/Parallel/ArrayCollection/DgElementCollection.hpp

```diff
@@ -0,0 +1,117 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+
+#include "Parallel/Algorithms/AlgorithmNodegroupDeclarations.hpp"
+#include "Parallel/ArrayCollection/CreateElementCollection.hpp"
+#include "Parallel/ArrayCollection/DgElementArrayMember.hpp"
+#include "Parallel/ArrayCollection/DgElementArrayMemberBase.hpp"
+#include "Parallel/ArrayCollection/Tags/ElementCollection.hpp"
+#include "Parallel/ArrayCollection/TransformPdalForNodegroup.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Local.hpp"
+#include "Parallel/ParallelComponentHelpers.hpp"
+#include "Parallel/Printf/Printf.hpp"
+#include "ParallelAlgorithms/Actions/TerminatePhase.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Parallel {
+/*!
+ * \brief A collection of DG elements on a node.
+ *
+ * The `PhaseDepActionList` is the PDAL that was used for the array
+ * approach. Some actions will require updating to support nodegroups if they
+ * haven't already been.
+ */
+template <size_t Dim, class Metavariables, class PhaseDepActionList>
+struct DgElementCollection {
+  /// \brief The `DgElementCollection` is currently a nodegroup.
+  ///
+  /// It should be possible to generalize this to work as a group too.
+  using chare_type = Parallel::Algorithms::Nodegroup;
+  /// \brief The metavariables
+  using metavariables = Metavariables;
+  /// \brief The simple tags necessary from option parsing.
+  using simple_tags_from_options = Parallel::get_simple_tags_from_options<
+      Parallel::get_initialization_actions_list<PhaseDepActionList>>;
+  /// \brief The type of the `Parallel::DgElementArrayMember` with all
+  /// template parameters.
+  using dg_element_array_member =
+      Parallel::DgElementArrayMember<Dim, Metavariables, PhaseDepActionList,
+                                     simple_tags_from_options>;
+  /// \brief The tag `Parallel::Tags::ElementCollection` of this
+  /// `DgElementCollection`
+  using element_collection_tag =
+      Parallel::Tags::ElementCollection<Dim, Metavariables, PhaseDepActionList,
+                                        simple_tags_from_options>;
+
+  /// @{
+  /// \brief Returns the derived `Parallel::DgElementArrayMember` from the
+  /// `Parallel::DgElementArrayMemberBase`
+  auto downcast(const Parallel::DgElementArrayMemberBase<Dim>& base)
+      -> const dg_element_array_member& {
+#ifdef SPECTRE_DEBUG
+    return dynamic_cast<const dg_element_array_member&>(base);
```

Comment:
what is the reason for doing the cast in this manner?

Reply:
I haven't needed these (thought I would) so just removing them.